src/core/main.js

```diff
@@ -271,8 +277,11 @@ class p5 {
 
         context.preload();
         this._runIfPreloadsAreDone();
+        this._afterPreload();
       } else {
+        this._beforeSetup();
```

Comment:
I think this branch only gets called of the sketch doesn't have a preload function, otherwise _runIfPreloadsAreDone will trigger setup to be called. To make sure we call these regardless of what triggers _setup(), maybe we can add these calls inside of _setup itself?